### ISSUE TITLE
Environment variable `POETRY_PYPI_TOKEN_<NAME>` not used by `poetry publish -r <name>`

### ISSUE THREAD

**Issue by esciara** [python-poetry/poetry#2210]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: 10.15.3 Mac OS X
- **Poetry version**: 1.0.5
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/esciara/dfc058348e597f8586d4f678cc163182

## Issue
After setting the following environment variables and verifying the system is correctly set:
```
$ export POETRY_REPOSITORIES_TESTPYPI_URL=https://test.pypi.org/legacy/
$ export POETRY_PYPI_TOKEN_TESTPYPI=<token>
$ env | grep POETRY
POETRY_REPOSITORIES_TESTPYPI_URL=https://test.pypi.org/legacy/
POETRY_PYPI_TOKEN_TESTPYPI=<token>
```
and building my package:
```
$ poetry build
Building mypackage (0.0.1)
 - Building sdist
 - Built mypackage-0.0.1.tar.gz

 - Building wheel
 - Built mypackage-0.0.1-py3-none-any.whl
```
I am asked for a username a password when attempting to publish, event thought the token is passed through the environment variable:
```
$ poetry publish -r testpypi

Publishing mypackage (0.0.1) to testpypi
Username:
```

I [checked the doc](https://python-poetry.org/docs/repositories/#configuring-credentials) (section on environment variables) and it should work.

## Workaround
The following workaround using username and password works:
```
$ export POETRY_REPOSITORIES_TESTPYPI_URL=https://test.pypi.org/legacy/
$ export POETRY_HTTP_BASIC_TESTPYPI_USERNAME=__token__
$ export POETRY_HTTP_BASIC_TESTPYPI_PASSWORD=<token>
```

**Comment by esciara** [comment]
Same problem occurs when using `pypi` through a straight `poetry publish`.

**Comment by esciara** [comment]
Can't reproduce the pb with the unit tests:
```
def test_publish_temp(app, environ, app_tester):
    os.environ["POETRY_REPOSITORIES_TESTREPO_URL"] = "<your_test_repo_url>"
    os.environ["POETRY_PYPI_TOKEN_TESTREPO"] = "<your_test_repo_token>"

    exit_code = app_tester.execute("publish -r testrepo")

    # should pass with 'assert exit_code == 0" or 
    # 'assert exit_code == 1" and an error message indicating a conflict 
    # if you are publishing the same package a second time
```
Provided you give valid `<your_test_repo_url>` and `<your_test_repo_token>`, this test will work (either success if package was never publish on `testrepo` or failure with error message indicating a conflict if you are publishing the same package a second time)

**Comment by arnavb** [comment]
Are there any updates on this? I'm experiencing the same on the Travis build for my package as well. The environment variable is just ignored.

**Comment by hukkinj1** [comment]
Here's a one-liner workaround that works for me:
```sh
poetry config http-basic.pypi "__token__" "${POETRY_PYPI_TOKEN_PYPI}"
```

**Comment by janhurst** [comment]
seeing the same behaviour as the OP. I was expecting to only need to set the `POETRY_PYPI_TOKEN_TESTPYPI` environment variable

**Comment by HacKanCuBa** [comment]
> Here's a one-liner workaround that works for me:
> 
> ```shell
> poetry config http-basic.pypi "__token__" "${POETRY_PYPI_TOKEN_PYPI}"
> ```

Doing `poetry config pypi-token.pypi "$POETRY_PYPI_TOKEN_PYPI"` as stated in the docs also works.

**Comment by tupui** [comment]
I actually had the issue and found that it was due to the URL path!

I had this, with the URL ending with `simple`. It is working fine for the `pyproject.toml`
```
poetry config repositories.mypypi https://pypi.mypypi.com/simple/
```
This does not work and removing `simple` does the trick.
```
poetry config repositories.mypypi https://pypi.mypypi.com
```

**Comment by whg517** [comment]
hi, guys.

I think the best practice is this:

first pass env:

```shell
export POETRY_HTTP_BASIC_TEST_USERNAME=__token__
export POETRY_HTTP_BASIC_TEST_PASSWORD=pypi-AgENdGVzdC5weX
# poetry not use `POETRY_REPOSITORIES_TEST`
export POETRY_REPOSITORIES_TEST=https://test.pypi.org/legacy/
```

config poetry repo. eg: ci action

```shell
poetry config repositories.test ${POETRY_REPOSITORIES_TEST}
```

publish

```
poetry publish -r test
```

**Comment by abn** [comment]
This should have been resolved with #9015.

**Comment by abn** [comment]
https://github.com/python-poetry/poetry/pull/9119 should add regression coverage for this issue.
